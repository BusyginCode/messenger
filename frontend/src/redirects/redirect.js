export default (path, { isServer, res }, history, condition = true) => {
  if (condition) {
    if (isServer) {
      res.writeHead(302, { Location: path })
      res.end()
    } else {
      history.replace(path)
    }
    return new Promise(() => {})
  } else {
    return Promise.resolve()
  }
}
