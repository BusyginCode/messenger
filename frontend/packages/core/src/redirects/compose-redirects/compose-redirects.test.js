import React from 'react'
import { shallow } from 'enzyme'
import { createContext } from '@messenger/testing'
import AuthRedirect from '../auth-redirect'
import redirect from './compose-redirects'

const Component = () => <div>Test</div>

describe('compose-redirects', () => {
  it('should correct', () => {
    const context = createContext()
    const Container = redirect([AuthRedirect()])
    expect(
      shallow(<Container route={{ routes: [{ component: Component }] }} />, context)
        .dive()
        .dive()
        .dive()
        .dive()
    ).toMatchSnapshot()
  })
})
