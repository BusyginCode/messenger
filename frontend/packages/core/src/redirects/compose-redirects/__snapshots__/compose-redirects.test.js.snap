// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compose-redirects should correct 1`] = `
<div>
  Test
</div>
`;
